### PR TITLE
New Resource : `azurerm_voice_services_communications_gateway_test_line`

### PR DESCRIPTION
**Pull request by sinbai**
Swagger Link: https://github.com/Azure/azure-rest-api-specs/blob/d632c45f406c48b5f72b7d031bf0742e46246c4b/specification/voiceservices/resource-manager/Microsoft.VoiceServices/stable/2023-01-31/voiceservices.json#L557

Test Result:
PASS: TestAccVoiceServicesTestLine_basic (234.97s)
PASS: TestAccVoiceServicesTestLine_requiresImport (308.26s)
PASS: TestAccVoiceServicesTestLine_complete (286.76s)
PASS: TestAccVoiceServicesTestLine_update (410.06s)

### CODE REVIEW

**Review by catriona-m — CHANGES_REQUESTED** [review]
Hi @sinbai thanks for this. I've left a couple of minor comments inline but overall this is looking pretty good. Thanks!

File: website/docs/r/voice_services_communications_gateway_test_line.html.markdown [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+}
+
+resource "azurerm_voice_services_communications_gateway" "example" {
+  name                = "example-vcg"
+  resource_group_name = azurerm_resource_group.example.name
+}
+
+resource "azurerm_voice_services_communications_gateway_test_line" "example" {
+  name                                     = "example-vtl"
+  location                                 = "West Europe"
```
[the remark this answers is not in the document]
Comment (sinbai) [reply website/docs/r/voice_services_communications_gateway_test_line.html.markdown]: Fixed.

File: internal/services/voiceservices/voice_services_communications_gateway_test_line_resource.go [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/location"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/voiceservices/2023-01-31/communicationsgateways"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/voiceservices/2023-01-31/testlines"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+type CommunicationsGatewayTestLineModel struct {
```
[the remark this answers is not in the document]
Comment (sinbai) [reply internal/services/voiceservices/voice_services_communications_gateway_test_line_resource.go]: Fixed.

**Comment by sinbai** [comment]
@catriona-m thanks for your feedback. In addition to fix the comments above, I also updated the setting file of teamcity  to update the `LocationConfiguration`, since the devtest subscription only supports one (`westcentralus` ) of the three regions to create resource group. Could you please take another look?

The test result as follows:
image

**Review by katbyte — APPROVED** [review]
LGTM ☎️

**Pull request merged**